datasetinsights/datasets/transformers/coco.py

```diff
@@ -167,3 +168,182 @@ def _categories(self):
+        self._kpt_def = ann_def.find_by_name(self.KPT_NAME)
+        try:
+            self._instance_segmentation_def = ann_def.find_by_name(
+                self.INSTANCE_SEGMENTATION_NAME
+            )
+        except NoRecordError as e:
+            logger.warning(
+                "Can't find instance segmentation annotations in the dataset. "
+                "The converted file will not contain instance segmentation."
+            )
+            logger.warning(e)
+            self._instance_segmentation_def = None
+
+        captures = Captures(
+            data_root=data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        )
+        self._bbox_captures = captures.filter(self._bbox_def["id"])
+        self._kpt_captures = captures.filter(self._kpt_def["id"])
+        if self._instance_segmentation_def:
+            self._instance_segmentation_captures = captures.filter(
+                self._instance_segmentation_def
+            )
+
+    def execute(self, output):
+        """Execute COCO Transformer
+        Args:
+            output (str): the output directory where converted dataset will
+              be stored.
+        """
+        self._copy_images(output)
+        self._process_instances(output)
+
+    def _copy_images(self, output):
+        image_to_folder = Path(output) / "images"
+        image_to_folder.mkdir(parents=True, exist_ok=True)
+        for _, row in self._bbox_captures.iterrows():
+            image_from = self._data_root / row["filename"]
+            if not image_from.exists():
+                continue
+            capture_id = uuid_to_int(row["id"])
+            image_to = image_to_folder / f"camera_{capture_id}.png"
+            shutil.copy(str(image_from), str(image_to))
+
+    def _process_instances(self, output):
+        output = Path(output) / "annotations"
+        output.mkdir(parents=True, exist_ok=True)
+        instances = {
+            "info": {"description": "COCO compatible Synthetic Dataset"},
+            "licences": [{"url": "", "id": 1, "name": "default"}],
+            "images": self._images(),
+            "annotations": self._annotations(),
+            "categories": self._categories(),
+        }
+        output_file = output / "keypoints.json"
+        with open(output_file, "w") as out:
+            json.dump(instances, out)
+
+    def _images(self):
+        images = []
+        for _, row in self._bbox_captures.iterrows():
+            image_file = self._data_root / row["filename"]
+            if not image_file.exists():
+                continue
+            with Image.open(image_file) as im:
+                width, height = im.size
+            capture_id = uuid_to_int(row["id"])
+            record = {
+                "file_name": f"camera_{capture_id}.png",
+                "height": height,
+                "width": width,
+                "id": capture_id,
+            }
+            images.append(record)
+
+        return images
+
+    def _annotations(self):
```

Comment:
The instance method `_annotations` and `_categories` might be the only difference in `COCOInstancesTransformer` and `COCOKeypointTransformer`. Can you use inheritance/composition to override the behavior for two different classes?